kad/kad.go

```diff
@@ -113,19 +115,24 @@ func Equal[K Key[K]](this, that NodeID[K]) bool {
 
 type Message interface{}
 
-type Request[K Key[K], A Address[A]] interface {
+type Request[K Key[K], N NodeID[K], A Address[A]] interface {
 	Message
 
 	// Target returns the target key and true, or false if no target key has been specfied.
 	Target() K
 
 	// EmptyResponse returns an empty response struct for this request message
 	// TODO: this is a weird patter, let's try to remove this.
-	EmptyResponse() Response[K, A]
+	EmptyResponse() Response[K, N, A]
 }
 
-type Response[K Key[K], A Address[A]] interface {
+type Response[K Key[K], N NodeID[K], A Address[A]] interface {
 	Message
 
-	CloserNodes() []NodeInfo[K, A]
+	CloserNodes() []NodeInfo[K, N, A]
+}
+
+type Protocol[K Key[K], N NodeID[K], A Address[A]] interface {
```

Comment:
What's the reasoning for adding N as a parameter to all these types?

Can't we just use the following

```Go
type Protocol[K Key[K],  A Address[A]] interface {	
	Get(ctx context.Context, to N NodeID[K], target K) (Response[K, A], error)
	Put(ctx context.Context, to N NodeID[K], record []byte) error
}
```